### ISSUE TITLE
is_dccp not closed under difference of convex/dccp functions.

### ISSUE THREAD

**Issue by se4u** [cvxgrp/dccp#13]
Hi CVX group, first of all thanks for the great work on CVXPY and DCCP. 

I am trying to test how well the dccp paradigm can be used for inference in a simple graphical model, and I tried to use the `cvx` `dccp` package to minimize the following objective

![image](https://cloud.githubusercontent.com/assets/2301364/17646480/42a12992-6199-11e6-80f6-43c1e28cc964.png)

Basically `x_i` are vectors that must lie in the positive simplex and `M_i` are predefined matrices.
Clearly both the components are convex as the following snippet shows.

```python
import dccp, cvxpy
objective_1 = sum([cvxpy.sum_squares(M * x)
                       for M, x in zip(Mlist, X)])
objective_2 = cvxpy.sum_squares(
        sum([(M * x) for M, x in zip(Mlist, X)]))
>>> print dccp.problem.is_dccp(cvxpy.Problem(cvxpy.Minimize(objective_1)))
     True
>>> print dccp.problem.is_dccp(cvxpy.Problem(cvxpy.Minimize(objective_2)))
     True
>>> print (cvxpy.Problem(cvxpy.Minimize(objective_2))).is_dcp()
     True
>>> print (cvxpy.Problem(cvxpy.Minimize(objective_1))).is_dcp()
     True
```

Since the objectives are convec their difference should be eligible for `dccp` but unfortunately that does not happen.
```python
>>> print dccp.problem.is_dccp(cvxpy.Problem(cvxpy.Minimize(objective_2 - objective_1)))
     False
```

As far as I can tell since I have not put any constraints the difference of two convex functions must be `dccp`. Am I overlooking something?

**Comment by SteveDiamond** [comment]
It's confusing given the name, but you don't write difference-of-convex functions directly in DCCP. See #11 . We need to update the documentation to make this much clearer. In your case you would write
```
import dccp, cvxpy
objective_1 = sum([cvxpy.sum_squares(M * x)
                       for M, x in zip(Mlist, X)])
objective_2 = cvxpy.sum_squares(
        sum([(M * x) for M, x in zip(Mlist, X)]))
t = cvxpy.Variable()
prob = cvxpy.Problem(cvxpy.Minimize(objective_1 - t), [ t == objective_2])
prob.solve(method='dccp')
```
You could also write ``[ t <= objective_2]``.

**Comment by se4u** [comment]
Thank you for the explanation. With the above mentioned change the problem now and with `tau=1` the optimizer is functions well. With default `tau` the optimizer returned an objective of `-inf` so I guess it was not paying any attention to the constraints with that setting.

As an aside, I modified the code in `dccp/problems.py` to skip the `is_dccp` test and run the procedure anyway and it worked fine. I was able to produce decent results without any tuning. It might be good to add a `force` flag to proceed with the procedure even if the objective is determined not to be `dccp` and issue a warning with more details. something like this...

```python
def dccp(self, max_iter=100, tau=0.005, mu=1.2, tau_max=1e8,
         solver=None, ccp_times=1, force=False, **kwargs):
    if not is_dccp(self):
        if force:
            warnings.warn("Not a DCCP problem. You have opted to force the procedure on a non-dccp problem. the results may not be good.")
        else:
            raise ValueError("Not a DCCP problem.")
```
